subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -393,63 +474,112 @@ private AttributeContainer concat(AttributeContainer configAttributes, Attribute
-                out.println();
+            for (Attribute<?> attribute : requested.keySet()) {
+                // If it's not in the matches, it's only in the requested attributes
+                if (bothAttributes.values().stream().map(d -> d.requested).noneMatch(Predicate.isEqual(attribute))) {
+                    requestedAttributes.add(attribute);
+                }
             }
-        }
 
-        private void writeFoundAttributes(StyledTextOutput out, AttributeContainer attributes, AttributeContainer requested, int maxAttributeLen, Set<Attribute<?>> matchedAttributes) {
-            for (Attribute<?> attribute : attributes.keySet()) {
-                Object actualValue = attributes.getAttribute(attribute);
-                AttributeMatchDetails match = match(attribute, actualValue, requested);
-                out.withStyle(Description).text("      " + StringUtils.rightPad(attribute.getName(), maxAttributeLen) + " = " + actualValue);
-                Attribute<?> requestedAttribute = match.requested;
-                if (requestedAttribute != null) {
-                    matchedAttributes.add(requestedAttribute);
+            List<String> header = new ArrayList<>(ImmutableList.of("Name", "Provided", "Requested"));
+            if (!selected) {
+                header.add("Compatibility");
+            }
+            List<TablePrinter.Row> rows = new ArrayList<>();
+            for (Attribute<?> attribute : providedAttributes) {
+                Object providedValue = attributes.getAttribute(attribute);
+                List<String> text = new ArrayList<>(ImmutableList.of(
+                    attribute.getName(),
+                    providedValue == null ? "" : providedValue.toString(),
+                    ""
+                ));
+                if (!selected) {
+                    text.add("Compatible");
                 }
+                rows.add(new TablePrinter.Row(text, Info));
+            }
+            for (Map.Entry<Attribute<?>, AttributeMatchDetails> entry : bothAttributes.entrySet()) {
+                Object providedValue = attributes.getAttribute(entry.getKey());
+                AttributeMatchDetails match = entry.getValue();
+                List<String> text = new ArrayList<>(ImmutableList.of(
+                    entry.getKey().getName(),
+                    providedValue == null ? "" : providedValue.toString(),
+                    String.valueOf(entry.getValue().requestedValue)
+                ));
+                StyledTextOutput.Style style;
                 switch (match.matchType) {
-                    case requested:
+                    case REQUESTED:
+                        style = Success;
                         break;
-                    case different_value:
-                        out.withStyle(Info).text(" (compatible with: " + match.requestedValue + ")");
+                    case DIFFERENT_VALUE:
```

Comment:
Should different value (compatible via rule) be highlighted with a WARN style?

Reply:
Why should different value be styled differently from a missing value on either side? I don't see a lot of value in that.